docs/developer-docs/backend/rust/10-timers.md

```diff
@@ -0,0 +1,237 @@
+ic-cdk-timers = "0.1"
+serde = { version = "1.0", features = ["derive"] }
+```
+
+Save the file. 
+
+## Update the Cargo.toml dependencies
+
+Since we made changes to the Cargo.toml file, run the following command to update the project's dependencies:
+
+```
+cargo update
+```
+
+## Declaring the canister interface
+
+Candid is an interface description language (IDL) for interacting with canisters running on the Internet Computer. Candid files provide a language-independent description of canister interfaces.
+
+To see details about the Candid interface description language syntax, see the [Candid guide](../candid/) or the [Candid crate documentation](https://docs.rs/candid/).
+
+Start by opening the `src/my_timers_backend/my_timers_backend.did` file in a code editor and replace its content with the following:
+
+```candid
+service : (nat64) -> {
+      "counter": () -> (nat64) query;
+}
+```
+
+This code establishes the following:
+
+- `service : (nat64) -> {...}` declares a new service which accepts a single integer argument.
+- `"counter": () -> (nat64) query` declares a canister query entry point named `counter`. The `counter` query takes no arguments `()` and returns an integer `(nat64)`.
+
+Save the file. 
+
+## Implementing the `counter` query
+
+In the previous step the `counter` query is declared as `"counter": () -> (nat64) query`. This step implements it.
+
+In the code editor, open the `src/my_timers_backend/src/lib.rs` file and replace its content with the following:
+
+```rust
+use std::sync::atomic::{AtomicU64, Ordering};
+
+static COUNTER: AtomicU64 = AtomicU64::new(0);
+
+#[ic_cdk::query]
+fn counter() -> u64 {
+    COUNTER.load(Ordering::Relaxed)
+}
+```
+
+This code establishes the following:
+
+- `static COUNTER: AtomicU64 = ...` defines a new global variable called `COUNTER`.
+- `#[ic_cdk::query]` marks the following `counter` function as a `query` entry point, so the function will be exported as `canister_query counter`.
+- `fn counter() -> u64 {...}` defines the query. Just like in the `.did` definition, it takes no arguments and returns `u64`.
+- `COUNTER.load(...)` loads and returns the global `COUNTER` value.
+
+## Implementing canister initialization
+
+In the previous step, the service is declared as `service : (nat64) -> {...}`. This step implements the canister initialization with an argument.
+
+In the code editor, open the `src/my_timers_backend/src/lib.rs` file and append the following:
+
+```rust
+[...]
+
+#[ic_cdk::init]
+fn init(timer_interval_secs: u64) {
+    let interval = std::time::Duration::from_secs(timer_interval_secs);
+    ic_cdk::println!("Starting a periodic task with interval {interval:?}");
+    ic_cdk_timers::set_timer_interval(interval, || {
+        COUNTER.fetch_add(1, Ordering::Relaxed);
+    });
+}
+```
+
+This code establishes the following:
+
+- `#[ic_cdk::init]` marks the following `init` function as a canister initialization method, so the function will be exported as `canister_init`.
```

Comment:
Saying what it's exported as will probably confuse rather than educate; users of ic-cdk do not need to know about `canister_init` or `canister_query counter`. The *meaning* of it, i.e. that it will be run when the canister is installed, is probably a better thing to put here and in the other two locations.